lib/joinserver/joinserver.go

```diff
@@ -116,3 +120,57 @@ func (s *JoinServiceGRPCServer) registerUsingIAMMethod(ctx context.Context, srv
 		Certs: certs,
 	}))
 }
+
+// RegisterUsingAzureMethod allows nodes and proxies to join the cluster using the
+// Azure join method.
+//
+// The server will generate a base64-encoded crypto-random challenge and
+// send it on the server stream. The caller is expected to respond on
+// the client stream with a RegisterUsingTokenRequest including a signed
+// attested data document with the challenge string. Finally, the signed
+// cluster certs are sent on the server stream.
+func (s *JoinServiceGRPCServer) RegisterUsingAzureMethod(srv proto.JoinService_RegisterUsingAzureMethodServer) error {
+	ctx := srv.Context()
+
+	timeout := s.clock.After(azureJoinRequestTimeout)
+
+	errCh := make(chan error, 1)
+	go func() {
+		errCh <- s.registerUsingAzureMethod(ctx, srv)
```

Comment:
Do you need to do this in the background? Why not just use context.WithTimeout to enforce a timeout?

Reply:
It's a workaround for canceling block `Send`/`Recv`s. It's explained in the comments for `RegisterUsingIAMMethod`; I have copied those comments over to avoid future confusion.